src/mac/mach.rs

```diff
@@ -0,0 +1,571 @@
+//! Contains various helpers to improve and expand on the bindings provided
+//! by `mach2`
+
+// Just exports all of the mach functions we use into a flat list
+pub use mach2::{
+    kern_return::{kern_return_t, KERN_SUCCESS},
+    port::mach_port_name_t,
+    task::{self, task_threads},
+    task_info,
+    thread_act::thread_get_state,
+    traps::mach_task_self,
+    vm::{mach_vm_deallocate, mach_vm_read, mach_vm_region_recurse},
+    vm_region::vm_region_submap_info_64,
+};
+
```

Comment:
The mach2 crate has a really annoying layout that I supposed they inherited from Apple.